### PR TITLE
[PUBDEV-7492] add validations for CoxPH parameters

### PR DESCRIPTION
**Pull request by honzasterba**

### CODE REVIEW

File: h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    heart = h2o.import_file(pyunit_utils.locate("smalldata/coxph_test/heart.csv"))
+    heart["age"] = heart["age"].ascharacter()
+    # validate start/stop columns
+    try:
+        coxph = H2OCoxProportionalHazardsEstimator(
+            start_column="XXX",
+            stop_column="YYY"
+        )
+        coxph.train(x=["surgery", "transplant", "year"], y="event", training_frame=heart)
+        assert False
```
Comment (maurever) [inline h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py:20]: Please, add error messages to all asserts in this test.

File: h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        assert "interactions: ZZZ not found in the training frame" in ex.args[0].exception_msg
+        assert "interactions_only: AAA not found in the training frame" in ex.args[0].exception_msg
+        assert "interaction_pairs: BBB not found in the training frame" in ex.args[0].exception_msg
+    # happy path
+    coxph = H2OCoxProportionalHazardsEstimator(
+        start_column="start",
+        stop_column="stop"
+    )
+    coxph.train(x=["surgery"], y="event", training_frame=heart)
+    assert True
```
Comment (maurever) [inline h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py:55]: Is this last assert necessary? It does not make sense to me. :)

Comment (honzasterba) [reply h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py:55]: removing

File: h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py (hunk 1 above)
Comment (honzasterba) [reply h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py:20]: done

Comment (maurever) [reply h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py:20]: I can't see an error message for `assert False`. :) However, If you think it is not important, I am ok with these changes.

Comment (honzasterba) [reply h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py]: forgot about those! added.

Comment (maurever) [reply h2o-py/tests/testdir_algos/coxph/pyunit_coxph_validations.py]: Perfect! Thank you @honzasterba!

**Pull request merged**